server.go

```diff
@@ -85,6 +119,8 @@ func (s *Server) Handle(ctx context.Context, conn net.Conn) {
 		if parseErr != nil {
 			writeErr = writer.WriteError(parseErr.Error())
 		} else {
+			s.monitorCommand(command)
```

Comment:
Let's move this out of the critical path.
We can have a server-wide monitor goroutine and provide the command to it through a dedicated channel here.
We should probably go with a buffered channel, so we are safe if the goroutine cannot catch up for a small time interval, and do not block on channel sending so we can apply some backpressure if the channel is full ( shouldn't happen often ).

Reply:
Right! This sounds like a better approach to me, too! I'll update the PR accordingly.